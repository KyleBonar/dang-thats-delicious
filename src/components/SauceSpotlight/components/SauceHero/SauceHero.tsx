import * as React from "react";
import { useSauceBySlug } from "../../../../utils/hooks/useSauceBySlug/useSauceBySlug";
import FlashMessage, {
  FlashMessageProps
} from "../../../FlashMessage/FlashMessage";

import {
  StyledSauceContainer,
  ImageContainer,
  StyleImg,
  StyledSauceInfoContainer,
  StyledH2
} from "./SauceHeroStyle";
import { ISauce } from "../../../../redux/sauces/types";

export interface SauceHeroProps {
  loading: boolean;
  sauce?: ISauce;
  error: FlashMessageProps;
}

const SauceHero: React.FunctionComponent<SauceHeroProps> = props => {
  // defaults
  const _loadingTxt = "loading...";
  const _noSauceTxt = "Could not find sauce!";
  const _defaultImagePath =
    "https://res.cloudinary.com/foryourthoughts/image/upload/v1565275178/sauces/ra1o7bsr9v2eurosoo5y.png";

  const { loading, sauce, error } = props;

  if (loading) {
    return <p>{_loadingTxt}</p>;
  }

  if (error.isVisible) {
    return <p>{error.text}</p>;
  }

  if (!sauce) {
    return <p>{_noSauceTxt}</p>;
  }

  return (
    <>
      {!sauce.isAdminApproved ?? (
        <FlashMessage
          isVisible={true}
          text={
            "This sauce has not been approved by an admin yet and, as a result, will not appear listed with the other sauces."
          }
          type="warning"
        />
      )}

      <StyledSauceContainer>
        <ImageContainer>
          {sauce.photo ? (
            <StyleImg src={`${sauce.photo}`} />
          ) : (
            <StyleImg src={_defaultImagePath} />
          )}
        </ImageContainer>
        <StyledSauceInfoContainer>
          <StyledH2>{sauce.name}</StyledH2>
          <p data-test-id="maker">
            <i>Maker:</i> {sauce.maker}
          </p>
          <p>
            <i>Description:</i> {sauce.description}
          </p>

          {sauce.ingredients && (
            <p data-test-id="ingredients">
              <i>Ingredients:</i> {sauce.ingredients}
            </p>
          )}

          {sauce.types && sauce.types.length > 0 && (
            <p data-test-id="type">
              <i>Type:</i> {sauce.types.join(", ")}
            </p>
          )}

          {sauce.shu && (
            <p data-test-id="shu">
              <i>SHU:</i> {sauce.shu} scoville
            </p>
          )}

          {sauce.country && (
            <p data-test-id="country">
              <i>Made in:</i> {sauce.country}
            </p>
          )}
        </StyledSauceInfoContainer>
      </StyledSauceContainer>
    </>
  );
};

export default SauceHero;
