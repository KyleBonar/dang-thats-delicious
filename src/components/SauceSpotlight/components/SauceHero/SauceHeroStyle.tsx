import styled from "styled-components";

export const StyledSauceContainer = styled.div`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  margin-bottom: 3em;
`;
StyledSauceContainer.displayName = "div";

export const ImageContainer = styled.div`
  max-width: 100%;
  padding: 0 5px;
  box-sizing: border-box;

  @media (min-width: ${props => props.theme.smToMd}) {
    width: 50%;
    padding: 0 15px;
  }
`;
ImageContainer.displayName = "div";

export const StyleImg = styled.img`
  width: 100%;
  box-sizing: border-box;
`;
StyleImg.displayName = "img";

export const StyledSauceInfoContainer = styled.div`
  width: 100%;
  box-sizing: border-box;
  padding: 0 5px;

  @media (min-width: ${props => props.theme.smToMd}) {
    width: 50%;
    padding: 0 15px;
  }
`;
StyledSauceInfoContainer.displayName = "div";

export const StyledH2 = styled.h2`
  margin-bottom: 16px;
`;
StyledH2.displayName = "h2";
