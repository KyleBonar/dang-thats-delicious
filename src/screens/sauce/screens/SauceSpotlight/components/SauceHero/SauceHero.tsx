import * as React from "react";
import { connect } from "react-redux";
import { AppState } from "../../../../../../redux/configureStore";
import { ISauce } from "../../../../../../redux/sauces/types";

import {
  StyledSauceContainer,
  StyledImageContainer,
  StyleImg,
  StyledSauceInfoContainer,
  StyledH2
} from "./SauceHeroStyle";

export interface SauceHeroProps {
  sauce?: ISauce; // This is sauce's slug that will have to be looked up
}

class SauceHero extends React.PureComponent<SauceHeroProps, any> {
  public constructor(props: SauceHeroProps) {
    super(props);
  }

  public render() {
    const { sauce } = this.props;

    return (
      <StyledSauceContainer>
        <StyledImageContainer>
          {sauce && sauce.photo ? (
            <StyleImg src={`${sauce.photo}`} />
          ) : (
            <StyleImg src="https://res.cloudinary.com/sfyt/image/upload/v1565275178/sauces/ra1o7bsr9v2eurosoo5y.png" />
          )}
        </StyledImageContainer>
        <StyledSauceInfoContainer>
          <StyledH2>{(sauce && sauce.name) || "Loading..."}</StyledH2>
          <p>
            <i>Maker:</i> {(sauce && sauce.maker) || "Loading..."}
          </p>
          <p>
            <i>Description:</i> {(sauce && sauce.description) || "Loading..."}
          </p>
          <p>
            <i>Ingredients:</i> {(sauce && sauce.ingredients) || "Loading..."}
          </p>
          <p>
            <i>Type:</i> {this.RenderType()}
          </p>
          <p>
            <i>Made in:</i> {(sauce && sauce.country) || "Loading..."}
          </p>
        </StyledSauceInfoContainer>
      </StyledSauceContainer>
    );
  }

  // Renders the type of sauce
  private RenderType = () => {
    const { sauce } = this.props;

    if (!sauce) return "Loading...";

    if (sauce.types && sauce.types.length > 0) return sauce.types.join(", ");

    return "N/A";
  };
}

const mapState2Props = (state: AppState) => {
  return {};
};

export default connect(mapState2Props)(SauceHero);
