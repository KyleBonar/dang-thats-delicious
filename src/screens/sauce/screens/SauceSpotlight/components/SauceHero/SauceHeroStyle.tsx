import styled from "styled-components";

export const StyledSauceContainer = styled.div`
  background-color: ${props => props.theme.formContainerBackgroundColor};
  padding: 1.5rem;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  margin-bottom: 3em;
`;

export const StyledImageContainer = styled.div`
  max-width: 100%;

  @media (min-width: ${props => props.theme.smToMd}) {
    width: 50%;
  }
`;

export const StyleImg = styled.img`
  width: 100%;
  box-sizing: border-box;
`;

export const StyledSauceInfoContainer = styled.div`
  font-family: AvenirNextReg;
  width: 100%;
  box-sizing: border-box;
  padding: 0 1em;

  @media (min-width: ${props => props.theme.smToMd}) {
    width: 50%;
  }
`;

export const StyledH2 = styled.h2`
  margin-top: 16px;
  margin-bottom: 16px;

  @media (min-width: ${props => props.theme.smToMd}) {
    margin-top: 51px;
  }
`;
