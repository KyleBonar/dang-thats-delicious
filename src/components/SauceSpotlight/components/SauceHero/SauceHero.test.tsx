import * as React from "react";
import * as enzyme from "enzyme";

import SauceHero from "./SauceHero";

describe("<SauceHero />", () => {
  it("renders", () => {
    // const wrapper = enzyme.
  });
});
