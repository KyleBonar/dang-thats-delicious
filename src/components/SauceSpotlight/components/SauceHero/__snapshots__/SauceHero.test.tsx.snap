// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SauceHero /> matches snapshot 1`] = `
<p>
  loading...
</p>
`;

exports[`<SauceHero /> matches snapshot 2`] = `
<p>
  loading...
</p>
`;

exports[`<SauceHero /> matches snapshot 3`] = `
<Fragment>
  <Component />
  <div>
    <div>
      <img
        src="http://Metz.ca/"
      />
    </div>
    <div>
      <h2>
        Mrs. Alec Dach
      </h2>
      <p
        data-test-id="maker"
      >
        <i>
          Maker:
        </i>
         
        Mrs. Kale Baumbach
      </p>
      <p>
        <i>
          Description:
        </i>
         
        facere fugit nemo libero consequuntur provident iusto
      </p>
      <p
        data-test-id="ingredients"
      >
        <i>
          Ingredients:
        </i>
         
        eum modi cum porro quasi nihil a
      </p>
      <p
        data-test-id="shu"
      >
        <i>
          SHU:
        </i>
         
        42
         scoville
      </p>
    </div>
  </div>
</Fragment>
`;

exports[`<SauceHero /> matches snapshot 4`] = `
<p>
  ut eum perspiciatis veritatis et accusamus atque
</p>
`;

exports[`<SauceHero /> matches snapshot 5`] = `
<p>
  loading...
</p>
`;

exports[`<SauceHero /> matches snapshot 6`] = `
<p>
  loading...
</p>
`;

exports[`<SauceHero /> matches snapshot 7`] = `
<Fragment>
  <Component />
  <div>
    <div>
      <img
        src="http://Grant.org/"
      />
    </div>
    <div>
      <h2>
        Miss Agnes Stracke
      </h2>
      <p
        data-test-id="maker"
      >
        <i>
          Maker:
        </i>
         
        Dr. Orville Harvey
      </p>
      <p>
        <i>
          Description:
        </i>
         
        qui eos velit magnam ea et id
      </p>
      <p
        data-test-id="shu"
      >
        <i>
          SHU:
        </i>
         
        121
         scoville
      </p>
    </div>
  </div>
</Fragment>
`;

exports[`<SauceHero /> matches snapshot 8`] = `
<p>
  loading...
</p>
`;

exports[`<SauceHero /> matches snapshot 9`] = `
<p>
  loading...
</p>
`;

exports[`<SauceHero /> matches snapshot 10`] = `
<p>
  quas quidem quisquam dicta veniam sunt autem
</p>
`;

exports[`<SauceHero /> matches snapshot 11`] = `
<p>
  loading...
</p>
`;

exports[`<SauceHero /> matches snapshot 12`] = `
<p>
  loading...
</p>
`;

exports[`<SauceHero /> matches snapshot 13`] = `
<p>
  loading...
</p>
`;

exports[`<SauceHero /> matches snapshot 14`] = `
<p>
  loading...
</p>
`;

exports[`<SauceHero /> matches snapshot 15`] = `
<p>
  loading...
</p>
`;

exports[`<SauceHero /> matches snapshot 16`] = `
<p>
  loading...
</p>
`;

exports[`<SauceHero /> matches snapshot 17`] = `
<p>
  loading...
</p>
`;

exports[`<SauceHero /> matches snapshot 18`] = `
<p>
  loading...
</p>
`;

exports[`<SauceHero /> matches snapshot 19`] = `
<p>
  loading...
</p>
`;

exports[`<SauceHero /> matches snapshot 20`] = `
<p>
  nihil omnis expedita dolores eos et similique
</p>
`;

exports[`<SauceHero /> matches snapshot 21`] = `
<Fragment>
  <div>
    <div>
      <img
        src="https://res.cloudinary.com/foryourthoughts/image/upload/v1565275178/sauces/ra1o7bsr9v2eurosoo5y.png"
      />
    </div>
    <div>
      <h2>
        Ms. Tate Nicolas
      </h2>
      <p
        data-test-id="maker"
      >
        <i>
          Maker:
        </i>
         
        Ms. Margarita Lindgren
      </p>
      <p>
        <i>
          Description:
        </i>
         
        deleniti tempora quibusdam aut doloribus et in
      </p>
      <p
        data-test-id="ingredients"
      >
        <i>
          Ingredients:
        </i>
         
        dolores ex qui quibusdam nostrum eos placeat
      </p>
      <p
        data-test-id="shu"
      >
        <i>
          SHU:
        </i>
         
        305
         scoville
      </p>
    </div>
  </div>
</Fragment>
`;

exports[`<SauceHero /> matches snapshot 22`] = `
<Fragment>
  <div>
    <div>
      <img
        src="http://Alf.ca/"
      />
    </div>
    <div>
      <h2>
        Mrs. Geoffrey Gutkowski
      </h2>
      <p
        data-test-id="maker"
      >
        <i>
          Maker:
        </i>
         
        Miss Tommie Bashirian
      </p>
      <p>
        <i>
          Description:
        </i>
         
        ut id enim ullam autem qui est
      </p>
      <p
        data-test-id="ingredients"
      >
        <i>
          Ingredients:
        </i>
         
        numquam et dolor sint impedit facere qui
      </p>
      <p
        data-test-id="country"
      >
        <i>
          Made in:
        </i>
         
        United Kingdom
      </p>
    </div>
  </div>
</Fragment>
`;

exports[`<SauceHero /> matches snapshot 23`] = `
<p>
  loading...
</p>
`;

exports[`<SauceHero /> matches snapshot 24`] = `
<p>
  loading...
</p>
`;

exports[`<SauceHero /> matches snapshot 25`] = `
<p>
  loading...
</p>
`;

exports[`<SauceHero /> matches snapshot 26`] = `
<p>
  loading...
</p>
`;

exports[`<SauceHero /> matches snapshot 27`] = `
<p>
  loading...
</p>
`;

exports[`<SauceHero /> matches snapshot 28`] = `
<p>
  temporibus vitae cum tempore nulla soluta autem
</p>
`;

exports[`<SauceHero /> matches snapshot 29`] = `
<p>
  et voluptatem minima vel recusandae ipsum exercitationem
</p>
`;

exports[`<SauceHero /> matches snapshot 30`] = `
<Fragment>
  <Component />
  <div>
    <div>
      <img
        src="http://Roma.biz/"
      />
    </div>
    <div>
      <h2>
        Mr. Rylee Pfeffer
      </h2>
      <p
        data-test-id="maker"
      >
        <i>
          Maker:
        </i>
         
        Ms. Roberto Will
      </p>
      <p>
        <i>
          Description:
        </i>
         
        nulla perferendis eligendi eos numquam voluptas dolores
      </p>
    </div>
  </div>
</Fragment>
`;

exports[`<SauceHero /> matches snapshot 31`] = `
<Fragment>
  <Component />
  <div>
    <div>
      <img
        src="https://res.cloudinary.com/foryourthoughts/image/upload/v1565275178/sauces/ra1o7bsr9v2eurosoo5y.png"
      />
    </div>
    <div>
      <h2>
        Mrs. Matilde Sauer
      </h2>
      <p
        data-test-id="maker"
      >
        <i>
          Maker:
        </i>
         
        Mrs. Kira Gerlach
      </p>
      <p>
        <i>
          Description:
        </i>
         
        id voluptates sed non eveniet sint et
      </p>
      <p
        data-test-id="shu"
      >
        <i>
          SHU:
        </i>
         
        417
         scoville
      </p>
    </div>
  </div>
</Fragment>
`;

exports[`<SauceHero /> matches snapshot 32`] = `
<p>
  et eum iste quibusdam tempore excepturi est
</p>
`;
